app/models/host.rb

```diff
@@ -500,6 +501,10 @@ def mac_addresses
     hardware.nil? ? [] : hardware.mac_addresses
   end
 
+  def thumbprint_sha1
+    ESXThumbPrint.new(ipaddress, authentication_userid, authentication_password).to_sha1
+  end
```

Comment:
:+1: much better, this should go to the vmware specific host model though: https://github.com/ManageIQ/manageiq-providers-vmware/blob/master/app/models/manageiq/providers/vmware/infra_manager/host_esx.rb since this doesn't make sense for all hosts.